test/unit-test/h264/h264_utest.c

```diff
@@ -199,38 +203,289 @@ void test_H264_Packetizer_AddFrame( void )
     Frame_t frame;
     Nalu_t nalusArray[ MAX_NALUS_IN_A_FRAME ];
     uint8_t pktBuffer[ MAX_H264_PACKET_LENGTH ];
-    uint32_t packetNumber = 0;
     uint32_t expectedPacketLength[] = { 2, 12, 4, 12, 12, 12, 12 };
 
     result = H264Packetizer_Init( &( ctx ),
                                   &( nalusArray[ 0 ] ),
                                   MAX_NALUS_IN_A_FRAME );
+
+    TEST_ASSERT_EQUAL( H264_RESULT_OK,
+                       result );
+
+    frame.pFrameData = pFrame;
+    frame.frameDataLength = frameLength;
+
+    result = H264Packetizer_AddFrame( &( ctx ),
+                                      &( frame ) );
+
+    TEST_ASSERT_EQUAL( H264_RESULT_OK,
+                       result );
+
+    pkt.pPacketData = &( pktBuffer[ 0 ] );
+    pkt.packetDataLength = MAX_H264_PACKET_LENGTH;
+
+    result = H264Packetizer_GetPacket( &( ctx ),
+                                       &( pkt ) );
+
+    size_t expectedPacketCount = sizeof( expectedPacketLength ) / sizeof( uint32_t );
+    for(size_t i = 0; i < expectedPacketCount; i++)
+    {
+        TEST_ASSERT_EQUAL( H264_RESULT_OK,
+                           result );
+        TEST_ASSERT_EQUAL( expectedPacketLength[i],
+                           pkt.packetDataLength );
+
+        pkt.pPacketData = &( pktBuffer[0] );
+        pkt.packetDataLength = MAX_H264_PACKET_LENGTH;
+
+        result = H264Packetizer_GetPacket( &( ctx ),
+                                           &( pkt ) );
+    }
+
+    TEST_ASSERT_EQUAL( H264_RESULT_NO_MORE_PACKETS,
+                       result );
```

Comment:
```suggestion
    size_t expectedPacketCount = sizeof( expectedPacketLength ) / sizeof( uint32_t );
    for(size_t i = 0; i < expectedPacketCount; i++)
    {
        pkt.pPacketData = &( pktBuffer[0] );
        pkt.packetDataLength = MAX_H264_PACKET_LENGTH;

        result = H264Packetizer_GetPacket( &( ctx ),
                                           &( pkt ) );

        TEST_ASSERT_EQUAL( H264_RESULT_OK,
                           result );
        TEST_ASSERT_EQUAL( expectedPacketLength[i],
                           pkt.packetDataLength );
    }

    /* After getting all content, H264Packetizer_GetPacket() returns H264_RESULT_NO_MORE_PACKETS. */
    pkt.pPacketData = &( pktBuffer[0] );
    pkt.packetDataLength = MAX_H264_PACKET_LENGTH;

    result = H264Packetizer_GetPacket( &( ctx ),
                                       &( pkt ) );

    TEST_ASSERT_EQUAL( H264_RESULT_NO_MORE_PACKETS,
                       result );
```
How about doing it in this way?

Reply:
done thankyou! this change makes it more readable.